doc/guide/glossary.rst

```diff
@@ -4,13 +4,116 @@ Glossary
 Here we define Flux-specific and general HPC and workload management terms
 used in our documentation that may not be familiar to all readers.
 
-  .. glossary::
+.. glossary::
 
-     initial program
-        A user-defined program, such as a batch script, launched on the first
-	node of a Flux instance.  Its purpose is to launch and monitor a
-	workload.  Once it is complete, the instance exits.
+  enclosing instance
+    The Flux instance that launched the current Flux instance as a job.
+
+  expedited
+    A job is said to be expedited if its :term:`urgency` is set to 31.
+    An expedited job's :term:`priority` is always set to the maximum value.
+
+  FSD
+    A common string representation of time duration, defined by
+    :doc:`rfc:spec_23`.  Example: ``2.5h``.
+
+  guest
+    A Flux user that is not the :term:`instance owner`.  Guests are only
+    allowed to run in a Flux instance configured for multi-user support,
+    normally a :term:`system instance`.
+
+  held
+    A job is said to be held if its :term:`urgency` is set to zero.  This
+    prevents it from being considered for scheduling until the urgency is
+    raised.
+
+  hostlist
+    A compact string representation of a list of hostnames, defined by
+    :doc:`rfc:spec_29`.  Example: ``fluke[0-127,130]``.
+
+  idset
+    A compact string representation of a set of non-negative integers,
+    defined by :doc:`rfc:spec_22`.  Example: ``2,4,6,1-100``.
+
+  IMP
+    The Independent Minister of Privilege.  The simple setuid root component
+    of Flux, from the flux-security project, that allows an
+    :term:`instance owner` to perform a limited set of tasks on behalf of a
+    :term:`guest` user in a multi-user Flux instance.
+
+  initial program
+    A user-defined program, such as a batch script, launched on the first
+    node of a Flux instance.  Its purpose is to launch and monitor a
+    workload.  Once it is complete, the instance exits.
```

Comment:
i know not your addition, but "purpose is to launch and monitor a workload" doesn't seem right?  "Typically it is used to launch ..."?

Reply:
Hm, "Its purpose is to launch and monitor a workload" works for me FWIW. (For me anyway, stating the purpose of the initial program gives more clarity than saying something like "Typically, ..." which implies ambiguity of what the initial program does) However, I don't feel strongly here and either could work.

Reply:
I guess in my mind someone can launch "anything" as the initial program, like a shell, but most people probably wouldn't do that.

Reply:
Yeah, so the more general statement of the initial program's purpose might work better? (E.g. `flux mini alloc` launches an interactive shell as the initial program, so a shell is actually common)